AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java

```diff
@@ -3531,12 +3643,58 @@ protected void showTagsDialog() {
         showDialogFragment(dialog);
     }
 
+    // init or reset api list
+    private void jsApiInit() {
+        mCardSuppliedApiVersion = "";
+        mCardSuppliedDeveloperContact = "";
+
+        for (int i = 0; i < sApiList.length; i++) {
+            mJsApiListMap.put(sApiList[i], false);
+        }
+    }
+
  /*
  Javascript Interface class for calling Java function from AnkiDroid WebView
 see card.js for available functions
  */
+    // list of api that can be accessed
+    private final String[] sApiList = {"toggleFlag", "markCard"};
+
     public class JavaScriptFunction {
 
+        private final Gson sGson = new Gson();
```

Comment:
```suggestion
        private final Gson mGson = new Gson();
```

Same notation comment as above - should be marked as member variable and usage altered to match